uxarray/helpers.py

```diff
@@ -63,10 +66,167 @@ def determine_file_type(filepath):
+
+    z : list, required
+        z-coordinate of all the nodes forming the face
+
+    type : str, optional
+        coordinate type, default is spherical, can be cartesian also.
+    """
+    area = 0  # set area to 0
+    order = 6
+    dG, dW = get_gauss_quadratureDG(order)
+
+    num_nodes = len(x)
+
+    # num triangles is two less than the total number of nodes
+    num_triangles = num_nodes - 2
+    # Using tempestremap GridElements: https://github.com/ClimateGlobalChange/tempestremap/blob/master/src/GridElements.cpp
+    # loop thru all triangles
+    for j in range(0, num_triangles):
+        node1 = [x[0], y[0], z[0]]
+        node2 = [x[j + 1], y[j + 1], z[j + 1]]
+        node3 = [x[j + 2], y[j + 2], z[j + 2]]
+        if (type == "spherical"):
+            node1 = spherical_to_cartesian_unit(node1)
+            node2 = spherical_to_cartesian_unit(node2)
+            node3 = spherical_to_cartesian_unit(node3)
+        for p in range(len(dW)):
+            for q in range(len(dW)):
+                dA = dG[p]
+                dB = dG[q]
+                jacobian = calculate_spherical_triangle_jacobian(
+                    node1, node2, node3, dA, dB)
+                area += dW[p] * dW[q] * jacobian
+    return area
+
+
+def calculate_spherical_triangle_jacobian(node1, node2, node3, dA, dB):
+    """Helper function for calculating face area."""
+    dF = [(1.0 - dB) * ((1.0 - dA) * node1[0] + dA * node2[0]) + dB * node3[0],
+          (1.0 - dB) * ((1.0 - dA) * node1[1] + dA * node2[1]) + dB * node3[1],
+          (1.0 - dB) * ((1.0 - dA) * node1[2] + dA * node2[2]) + dB * node3[2]]
+
+    dDaF = [(1.0 - dB) * (node2[0] - node1[0]),
+            (1.0 - dB) * (node2[1] - node1[1]),
+            (1.0 - dB) * (node2[2] - node1[2])]
+
+    dDbF = [
+        -(1.0 - dA) * node1[0] - dA * node2[0] + node3[0],
+        -(1.0 - dA) * node1[1] - dA * node2[1] + node3[1],
+        -(1.0 - dA) * node1[2] - dA * node2[2] + node3[2]
+    ]
+
+    dInvR = 1.0 / math.sqrt(dF[0] * dF[0] + dF[1] * dF[1] + dF[2] * dF[2])
+
+    dDaG = [
+        dDaF[0] * (dF[1] * dF[1] + dF[2] * dF[2]) - dF[0] *
+        (dDaF[1] * dF[1] + dDaF[2] * dF[2]),
+        dDaF[1] * (dF[0] * dF[0] + dF[2] * dF[2]) - dF[1] *
+        (dDaF[0] * dF[0] + dDaF[2] * dF[2]),
+        dDaF[2] * (dF[0] * dF[0] + dF[1] * dF[1]) - dF[2] *
+        (dDaF[0] * dF[0] + dDaF[1] * dF[1])
+    ]
+
+    dDbG = [
+        dDbF[0] * (dF[1] * dF[1] + dF[2] * dF[2]) - dF[0] *
+        (dDbF[1] * dF[1] + dDbF[2] * dF[2]),
+        dDbF[1] * (dF[0] * dF[0] + dF[2] * dF[2]) - dF[1] *
+        (dDbF[0] * dF[0] + dDbF[2] * dF[2]),
+        dDbF[2] * (dF[0] * dF[0] + dF[1] * dF[1]) - dF[2] *
+        (dDbF[0] * dF[0] + dDbF[1] * dF[1])
+    ]
+
+    dDenomTerm = dInvR * dInvR * dInvR
+
+    dDaG[0] *= dDenomTerm
+    dDaG[1] *= dDenomTerm
+    dDaG[2] *= dDenomTerm
+
+    dDbG[0] *= dDenomTerm
+    dDbG[1] *= dDenomTerm
+    dDbG[2] *= dDenomTerm
```

Comment:
If these are numpy arrays, could shorten to
```suggestion
    dDaG *= dDenomTerm
    
    dDbG *= dDenomTerm
```